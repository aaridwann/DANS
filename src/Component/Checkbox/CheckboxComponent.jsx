import React from "react";

function CheckboxComponent({ title, value, onChange }) {
  return (
    <div className="flex">
      <input
        onChange={onChange}
        // checked={onChange}
        value={value}
        type="checkbox"
        id="choose-me"
        className="peer hidden"
      />
      <label
        htmlFor="choose-me"
        className="select-none cursor-pointer rounded-lg border-2 border-gray-200 py-3 px-6 font-bold text-gray-200 transition-colors duration-200 ease-in-out peer-checked:bg-cyan-500 peer-checked:text-white peer-checked:border-none "
      >
        {title}
      </label>
    </div>
  );
}

export default CheckboxComponent;
