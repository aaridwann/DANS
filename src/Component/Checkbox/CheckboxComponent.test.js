import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CheckboxComponent from "./CheckboxComponent";

describe("CheckboxComponent", () => {
  const props = {
    title: "Checkbox title",
    value: false,
    onChange: jest.fn(),
  };

  it("Snapshot testing", () => {
    expect(render(<CheckboxComponent />)).toMatchSnapshot();
  });

  it("renders correctly with props", () => {
    const { getByLabelText } = render(<CheckboxComponent {...props} />);
    const checkboxInput = getByLabelText(props.title);
    expect(checkboxInput.checked).toBe(props.value);
  });

  it("calls onChange when clicked", () => {
    const { getByLabelText } = render(<CheckboxComponent {...props} />);
    const checkboxInput = getByLabelText(props.title);
    fireEvent.click(checkboxInput);
    expect(props.onChange).toHaveBeenCalled();
  });
});
