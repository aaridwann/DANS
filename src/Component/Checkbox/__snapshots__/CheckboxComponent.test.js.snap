// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxComponent Snapshot testing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex"
      >
        <input
          class="peer hidden"
          id="choose-me"
          type="checkbox"
          value=""
        />
        <label
          class="select-none cursor-pointer rounded-lg border-2 border-gray-200 py-3 px-6 font-bold text-gray-200 transition-colors duration-200 ease-in-out peer-checked:bg-cyan-500 peer-checked:text-white peer-checked:border-none "
          for="choose-me"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex"
    >
      <input
        class="peer hidden"
        id="choose-me"
        type="checkbox"
        value=""
      />
      <label
        class="select-none cursor-pointer rounded-lg border-2 border-gray-200 py-3 px-6 font-bold text-gray-200 transition-colors duration-200 ease-in-out peer-checked:bg-cyan-500 peer-checked:text-white peer-checked:border-none "
        for="choose-me"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
